File: src/main/java/no/nav/tag/sykefravarsstatistikk/api/domain/Fnr.java
package no.nav.tag.sykefravarsstatistikk.api.domain;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.Value;

@Value
public class Fnr implements Identifikator {

    private final String verdi;

    public Fnr(String verdi) {
        if (!erGyldigFnr(verdi)) {
            throw new RuntimeException("Ugyldig fødselsnummer. Må bestå av 11 tegn.");
        }
        this.verdi = verdi;
    }

    public static boolean erGyldigFnr(String fnr) {
        return fnr.matches("^[0-9]{11}$");
    }

    @JsonValue
    @Override
    public String asString() {
        return verdi;
    }
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/modell/Fnr.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.modell

import com.fasterxml.jackson.annotation.JsonValue
import lombok.Value

@Value
class Fnr(val verdi: String) {

    init {
        if (!verdi.matches("^[0-9]{11}$".toRegex())) {
            throw UgyldigFnrException()
        }
    }

    @JsonValue
    fun asString(): String {
        return verdi
    }

    class UgyldigFnrException : RuntimeException("Ugyldig fødselsnummer. Må bestå av 11 tegn.")
}